### ISSUE TITLE
(Bug) MarketCalendar.open_at_time recognizes timezone differences for time but not for date

### ISSUE THREAD

**Issue by leonarduschen** [rsheftel/pandas_market_calendars#112]
Name: pandas-market-calendars
Version: 1.4.2

My first issue submission, apologies in advance for any formatting mistakes.

The title says it all. Any timezone difference at "hours" level is recognized but at "date" level is not.

Code to reproduce:
```
import pandas as pd
import pandas_market_calendars as mcal

hkex = mcal.get_calendar('HKEX')
schedule = hkex.schedule('2020-05-04', '2020-05-08', tz='America/New_York)

tf0 = pd.Timestamp('2020-05-04 03:00:00', tz='UTC')
tf1 = tf0.tz_convert('America/New_York')
tf2 = tf0.tz_convert('Hongkong')

print(tf0, hkex.open_at_time(schedule, tf0))
print(tf1, hkex.open_at_time(schedule, tf1))
print(tf2, hkex.open_at_time(schedule, tf2))
```

Output:
```
2020-05-04 03:00:00+00:00 True
2020-05-03 23:00:00-04:00 False
2020-05-04 11:00:00+08:00 True
```

All three timestamps are equivalent and we would expect all three to output True.  This is somewhat related to #87 as the plan is to take into account timezone in the local system (i.e. if system timezone is at New York but trading at HKEX). I have found what's causing the issue in the source code:

```
def open_at_time(schedule, timestamp, include_close=False):
    date = pd.Timestamp(timestamp.date())
    if date in schedule.index:
    ...
```
`date` refers to the 'raw' date of the timestamp argument but `schedule.index` is **always** in UTC as defined in the `MarketCalendar.schedule` method: `_all_days = self.valid_days(start_date, end_date)`

The issue can be fixed by simply changing `date` to be time aware. I'll make a PR if that sounds good.

Note: as of now the same issue also applies to `MarketCalendar.schedule`. It fails to take into account timezone when taking timezone aware timestamps for `start_date` and `end_date` arguments. This is perhaps a smaller issue but the fix is very similar and I can make a PR for that too if you'd like.

**Comment by rsheftel** [comment]
If you could make a PR with tests to confirm it works and futures changes do not revert your work that would be great, thank you.

**Comment by rsheftel** [comment]
Fixed in 9db441bd81f4a1aaa3bc52c4c22ad112df46cbac
